src/combinatorics/adjacency_matrix.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author:  Aaron Anderson, Jalex Stark.
+-/
+import linear_algebra.matrix
+import data.rel
+import combinatorics.simple_graph
+
+open_locale classical big_operators
+open finset matrix simple_graph
+noncomputable theory
+
+universes u v
+variables {α : Type u} [fintype α] [decidable_eq α]
+variables (R : Type v) [semiring R]
+
+namespace simple_graph
+
+variables (G : simple_graph α) (R)
+
+/-- The matrix $A$ such that $A i j = 1$ if $i$ and $j$ are adjacent, and otherwise $A i j = 0$-/
+def adjacency_matrix : matrix α α R := λ i j, ite (G.adj i j) 1 0
```

Comment:
We often write matrix definitions using the equation compiler, to hint that applying it to one argument, such as `G.adjacency_matrix R i`, doesn't really make sense.
```suggestion
def adjacency_matrix : matrix α α R
| i j := ite (G.adj i j) 1 0
```